ctms/schemas/bulk.py

```diff
@@ -0,0 +1,54 @@
+import base64
+from datetime import datetime, timezone
+from typing import Literal, Optional, Union
+
+from pydantic import validator
+
+from .base import ComparableBase
+
+BLANK_VALS = [None, "", "null"]  # TODO: Should "null" be allowed?
+
+
+class BulkRequestSchema(ComparableBase):
+    """A Bulk Read Request."""
+
+    start_time: datetime
+
+    end_time: Optional[Union[datetime, Literal[""]]] = None
+
+    @validator("end_time")
+    def end_time_must_not_be_blank(cls, value):  # pylint: disable=E0213
+        if value in BLANK_VALS:
```

Comment:
After reading [Validate Always](https://pydantic-docs.helpmanual.io/usage/validators/#validate-always), I'm not sure if this is called when value is ``None``. We may only be looking for blank strings here.

Reply:
During testing I think None got here, because the fields were Optional?

Reply:
Perfect, thanks for confirming.